lib/Relationship.php

```diff
@@ -602,7 +602,7 @@ public function create_association(Model $model, $attributes=array(), $guard_att
 	public function load_eagerly($models=array(), $attributes=array(), $includes, Table $table)
 	{
 		$this->set_keys($table->class->name);
-		$this->query_and_attach_related_models_eagerly($table,$models,$attributes,$includes,$this->foreign_key, $table->pk);
+		$this->query_and_attach_related_models_eagerly($table,$models,$attributes,$includes,$this->foreign_key, $this->primary_key);
```

Comment:
To use the primary key on the association, and not the table.  Primary key can be explicitly set in the association, and that key might be different than the table's primary key.  I believe if the association doesn't have a primary key, the table's primary key is used.